### ISSUE TITLE
Support for endless method definitions

### ISSUE THREAD

**Issue by gemmaro** [ruby/rdoc#885]
It seems that RDoc does not yet support the endless method at this time.

Here is a small example:

`lib/main.rb`:

```ruby
class AAA
  def aaa = :aaa
end

class BBB; end
```

then `$ rdoc` produces following document:

* Class and Module Index
  * `AAA`
  * `AAA::BBB` (if correct, it is `BBB`)
* class `AAA`'s public instance methods: the following screenshot

![Screen Shot 2022-05-23 at 12 28 00](https://user-images.githubusercontent.com/29575029/169737746-79694d92-1c59-4861-b3b2-f4213005b929.png)

In other words, the end of the range of method `aaa` is misidentified as the `end` of class `AAA`.

This doesn't happen if the method isn't defined as the endless method.

Using following versions:

```shell-session
$ ruby -v
ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-darwin21]

$ rdoc -v
6.4.0
```

**Comment by halostatue** [comment]
This is fixed with `RDOC_USE_PRISM_PARSER`.

**Comment by gemmaro** [comment]
Thank you for letting me know!  I've confirmed that everything is working as expected, so I'll close this issue now.
